api/v1alpha1/ippool_webhook.go

```diff
@@ -112,26 +114,50 @@ func (c *IPPool) checkPoolBonds(old *IPPool) ([]IPAddressStr, []IPAddressStr) {
 	return allocationOutOfBonds, inUseOutOfBonds
 }
 
-func (c *IPPool) isAddressInBonds(address IPAddressStr) bool {
-	inBonds := false
+func (c *IPPool) isAddressInBounds(address IPAddressStr) bool {
+	ip, err := netip.ParseAddr(string(address))
+	if err != nil {
+		return false
+	}
+
 	for _, pool := range c.Spec.Pools {
-		if inBonds {
-			break
+		if pool.Start != nil {
+			startIP, err := netip.ParseAddr(string(*pool.Start))
+			if err != nil {
+				// skip this invalid pool, as the validation error should be caught somewhere else
+				continue
+			}
+			if startIP.Compare(ip) > 0 {
+				continue
+			}
 		}
-		index := 0
-		for !inBonds {
-			allocatedAddress, err := GetIPAddress(pool, index)
+
+		if pool.End != nil {
+			endIP, err := netip.ParseAddr(string(*pool.End))
 			if err != nil {
-				break
+				// skip this invalid pool, as the validation error should be caught somewhere else
+				continue
 			}
-			index++
-			if allocatedAddress == address {
-				inBonds = true
-				break
+			if endIP.Compare(ip) < 0 {
+				continue
 			}
 		}
+
+		if pool.Subnet != nil {
+			_, subnet, err := net.ParseCIDR(string(*pool.Subnet))
```

Comment:
looks like my comment didn't land. Not a fan of empty vars `_`, we are constantly updating the linters and this might be an issue in near future.

Reply:
I think the linters would complain if we put something other than `_`? This is a nice way of saying we do not use that variable.

Reply:
Actually yes, its the other way around, sorry for mixing up